kore/src/Kore/Strategies/Goal.hs

```diff
@@ -164,7 +188,7 @@ class Goal goal where
         :: goal
         -> [goal]
         -> [Rule goal]
-        -> [Strategy (Prim goal)]
+        -> Stream (Strategy (Prim goal))
```

Comment:
Why this change?

Reply:
This was for generalizing `verifyClaimStep`. I think there were 2 ways to solve this:
  1. assume `strategy` returns an infinite list and then `verifyClaimStep` can extract steps from it
  2.  add `strategyFirstStep` and `strategySecondStep` to the class but that would assume all claim types have at maximum 2 different steps (but it's very possible that we wouldn't want something more general anyway)

I opted for 1 since it seemed more general and easier to refactor. I can do 2 if that makes more sense, or maybe there is a better way.

Reply:
I agree, returning an infinite stream as the strategy makes sense!